File: backend/app/api/routes/share.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.api.deps import get_current_user, get_db
from app.models.system import User
from app.schemas.share import ShareLinkResponse
from app.services import auth_service

router = APIRouter(tags=["share"], dependencies=[Depends(get_current_user)])


@router.get("/api/share-link", response_model=ShareLinkResponse)
def get_share_link(
    db: Session = Depends(get_db), current_user: User = Depends(get_current_user)
) -> ShareLinkResponse:
    token = auth_service.get_or_create_share_token(db, current_user)
    return ShareLinkResponse(token=token)


@router.post("/api/share-link/regenerate", response_model=ShareLinkResponse)
def regenerate_share_link(
    db: Session = Depends(get_db), current_user: User = Depends(get_current_user)
) -> ShareLinkResponse:
    token = auth_service.regenerate_share_token(db, current_user)
    return ShareLinkResponse(token=token)

File: frontend/api/share.ts
import { apiClient } from "./client";

export async function getShareLink(): Promise<string> {
  const response = await apiClient.get<{ token: string }>("/api/share-link");
  return response.data.token;
}

export async function regenerateShareLink(): Promise<string> {
  const response = await apiClient.post<{ token: string }>("/api/share-link/regenerate");
  return response.data.token;
}
